File: solutions/python3/DesignAStackWithIncrementOperation.py
class CustomStack:

    def __init__(self, maxSize: int):
        self.stack = []
        self.max = maxSize

    def push(self, x: int) -> None:
        if self._is_full():
            return
        self.stack.append(x)
        
    def pop(self) -> int:
        if self._is_empty():
            return -1
        return self.stack.pop()

    def increment(self, k: int, val: int) -> None:
        for i in range(min(k, len(self.stack))):
            self.stack[i] += val
    
    def _is_full(self):
        return len(self.stack) == self.max
        
    def _is_empty(self):
        return len(self.stack) == 0

# Your CustomStack object will be instantiated and called as such:
# obj = CustomStack(maxSize)
# obj.push(x)
# param_2 = obj.pop()
# obj.increment(k,val)

File: solutions/java/DesignAStackWithIncrementOperation.java
class CustomStack {

    int max;
    Stack<Integer> stack;

    public CustomStack(int maxSize) {
        max = maxSize;
        stack = new Stack<>();
    }
    
    public void push(int x) {
        if (!isFull())
            stack.add(x);
    }
    
    public int pop() {
        if (stack.isEmpty())
            return -1;
        return stack.pop();
    }
    
    public void increment(int k, int val) {
        int min = Math.min(k, stack.size());
        for (int i = 0; i < min; i++)
            stack.set(i, stack.get(i) + val);
    }

    private boolean isFull(){
        return max == stack.size();
    }

}

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack obj = new CustomStack(maxSize);
 * obj.push(x);
 * int param_2 = obj.pop();
 * obj.increment(k,val);
 */